core/domain/voiceover_services.py

```diff
@@ -0,0 +1,291 @@
+
+    Args:
+        user_id: str. The user ID of the reviewer.
+
+    Returns:
+        list(BaseVoiceoverApplication). A list of voiceover application which
+        the given user can review.
+    """
+    voiceover_application_models = (
+        suggestion_models.GeneralVoiceoverApplicationModel
+        .get_reviewable_voiceover_applications(user_id))
+
+    return [
+        _get_voiceover_application_from_model(model) for model in (
+            voiceover_application_models)]
+
+
+def get_user_submitted_voiceover_applications(user_id, status=None):
+    """Returns a list of voiceover application submitted by the given user which
+    are currently in the given status.
+
+    Args:
+        user_id: str. The id of the user.
+        status: str|None. The status of the voiceover application.
+
+    Returns:
+        BaseVoiceoverApplication). A list of voiceover application which are
+        submitted by the given user.
+    """
+    voiceover_application_models = (
+        suggestion_models.GeneralVoiceoverApplicationModel
+        .get_user_voiceover_applications(user_id, status))
+
+    return [
+        _get_voiceover_application_from_model(model) for model in (
+            voiceover_application_models)]
+
+
+def accept_voiceover_application(voiceover_application_id, reviewer_id):
+    """Accept the voiceover application of given voiceover application id.
+
+    Args:
+        voiceover_application_id: str. The is of the voiceover application which
+            need to be accepted.
+        reviewer_id: str. The user ID of the reviewer.
+    """
+    voiceover_application = get_voiceover_application_by_id(
+        voiceover_application_id)
+    if reviewer_id == voiceover_application.author_id:
+        raise Exception(
+            'Applicants are not allowed to review their own '
+            'voiceover application.')
+
+    reviewer = user_services.UserActionsInfo(user_id=reviewer_id)
+
+    voiceover_application.accept(reviewer_id)
+
+    _save_voiceover_application(voiceover_application)
+
+
+    if voiceover_application.target_type == feconf.ENTITY_TYPE_EXPLORATION:
+        rights_manager.assign_role_for_exploration(
+            reviewer, voiceover_application.target_id,
+            voiceover_application.author_id, rights_manager.ROLE_VOICE_ARTIST)
+        opportunity_services.update_exploration_voiceover_opportunities(
+            voiceover_application.target_id,
+            voiceover_application.language_code)
+        opportunity = (
+            opportunity_services.get_exploration_opportunity_summaries_by_ids([
+                voiceover_application.target_id]))
+        email_manager.send_accepted_voiceover_application_email(
+            voiceover_application.author_id, opportunity['0'].chapter_title,
+            voiceover_application.language_code)
+    # Need to reject all other voiceover application for the same entity?
+    # Add notification?
```

Comment:
A notification could be good.

Reply:
I've created an issue at #7969 and added a todo around that line! (Community dashboard team will work on "extra features" later.)

Reply:
Please remove the code comment!

Reply:
(It's already done!)